aws/protocol/json/decoder_util.go

```diff
@@ -119,5 +104,25 @@ func DiscardUnknownField(decoder *json.Decoder) error {
 		}
 	}
 
-	return err
+	return nil
+}
+
+// GetSmithyGenericAPIError returns smithy generic api error and an error interface.
+// Takes in json decoder, and error Code string as args. The function retrieves error message
+// and error code from the decoder body. If errorCode of length greater than 0 is passed in as
+// an argument, it is used instead.
+func GetSmithyGenericAPIError(decoder *json.Decoder, errorCode string) (smithy.GenericAPIError, error) {
+	errorType, message, err := GetErrorInfo(decoder)
+	if err != nil {
+		return smithy.GenericAPIError{}, err
+	}
+
+	if len(errorCode) == 0 {
+		errorCode = SanitizeErrorCode(errorType)
+	}
```

Comment:
It looks like the error code value is sanitized in a few spots. Not a big deal, but this could probably be cleaned up a bit, Maybe the utility that does the GetErrorInfo might be a better spot.